### PR TITLE
fix(mongodb-query-parser): missing `bson` peer dependency

### LINKED ISSUE #471: chore(mongodb-query-parser): missin `bson` dependency
#### Current Behavior

```console
mongodb-query-parser@npm:4.2.4 doesn't provide bson to @mongodb-js/shell-bson-parser
```

#### Code/Gist

- `yarn init`
- `yarn set version berry`
- `yarn add mongodb-query-parser`

#### Expected Behavior/Code

No warning on _yarn_ console

#### Possible Solution

Add `bson` as dependency or peer dependency.

`mongodb-query-parser` only use `bson` in [stringift.ts](https://github.com/mongodb-js/devtools-shared/blob/main/packages/query-parser/src/stringify.ts) for _types_. 

Alternatives:
- import _types_ from another library in dependencies and add `bson` as peer dependency
- add `bson` as peer dependency (I don't know if the import for _types_ can be ignored) or dependency

### PR DESCRIPTION
**Pull request by rtritto**
Fix #471

## Checklist
- [x] I have signed the Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)
- [x] New tests and/or benchmarks are included
- [x] Documentation is changed or added

### CODE REVIEW

File: packages/query-parser/package.json [hunk 1]
```diff
@@ -69,5 +69,8 @@
     "prettier": "^2.3.2",
     "sinon": "^9.2.3",
     "typescript": "^5.0.4"
+  },
+  "peerDependencies": {
+    "bson": "^6.8.0"
```
Comment (Anemy) [inline packages/query-parser/package.json:74]: Thanks for opening the pr!
Should we match the `peerDependencies` of `shell-bson-parser` here? https://github.com/mongodb-js/devtools-shared/blob/ca8aea2deb73026172732c1cf5cd8d36208ec440/packages/shell-bson-parser/package.json#L51
```suggestion
    "bson": "^4.6.3 || ^5 || ^6"
```

Comment (rtritto) [reply packages/query-parser/package.json]: Thanks, I did the fix

**Comment by rtritto** [comment]
FYI @addaleax

**Comment by Anemy** [comment]
Thanks for opening the pr!

**Pull request merged**